lib/electric_slide/call_queue.rb

```diff
@@ -241,6 +241,8 @@ def remove_agent(agent, extra_params = {})
     # Checks to see if any callers are waiting for an agent and attempts to connect them to
     # an available agent
     def check_for_connections
+      # Ensure there are no nil objects in the call queue before trying to connect
+      @queue.compact!
```

Comment:
I cannot reproduce the issue on a small scale, only when we have more than 50 agents. The outbound call is valid and active when it is queued but it seems that by the time the loop checking for connections pops that call off the queue the call is then nil. My only thought is that the call dies and becomes nil somewhere between it being queued and getting connection, yet the connection is happening before the call is removed.

My thinking for this is that call_waiting? returns true still meaning the queue has a nil object on it, rather than it being empty. I have been battling to find the exact cause. This bandaid does solve the issue, but I agree it is not an ideal solution.